### PR TITLE
feat(daemon): route bare loongclaw into welcome

### LINKED ISSUE #400: [Feature]: Change default loongclaw command from Demo to onboarding flow
### Surface
Daemon / CLI / install

### Problem statement
When new users run loongclaw without arguments for the first time, they see confusing internal debug output instead of a guided onboarding experience:

```
task dispatched via EmbeddedPi: {"adapter":"pi-local","objective":"summarize flaky test clusters","task":"task-bootstrap-01"}
connector dispatch: {"connector":"webhook","operation":"notify","payload":{"channel":"ops-alerts","message":"task complete"}}
```

Image

This output:
- Looks like internal debug logs, not user-facing content
- Does not guide users toward the recommended first step (loongclaw onboard)
- Conflicts with the README which emphasizes onboarding as the primary entry point
- Wastes the critical first-impression moment for new users

### Proposed solution
Change the default command (when no subcommand is provided) from Demo to a new Welcome flow that:

1. **Detects config state** - Check if ~/.loongclaw/config.toml exists
2. **If no config exists** - Auto-start the interactive onboard flow
3. **If config exists** - Show a friendly welcome banner with:
   - Quick commands reference (ask, chat, doctor)
   - Version info
   - One-line hint for help (loongclaw --help)

**Implementation sketch:**
- Modify crates/daemon/src/main.rs:41 - Change unwrap_or(Commands::Demo) to unwrap_or(Commands::Welcome)
- Create new Commands::Welcome variant or reuse Onboard with smart defaults
- Add config existence check before routing decision

### Non-goals / out of scope
- Removing the Demo command entirely (keep it as explicit loongclaw demo)
- Changing the onboard flow itself
- Auto-creating config without user consent

### Alternatives considered
1. **Keep Demo as default** - Rejected because it does not help new users get started
2. **Show static help message** - Less engaging than directly starting onboarding
3. **Detect first-run via marker file** - Adds complexity; config existence is sufficient

### Acceptance criteria
- [ ] Running loongclaw without args starts onboard flow when no config exists
- [ ] Running loongclaw without args shows welcome message when config exists
- [ ] loongclaw demo still works as explicit command
- [ ] No breaking changes to existing CLI behavior (explicit commands unchanged)

### Policy / security / breaking sensitivity
No, this looks routine

### Impact
**Affected users:** All new LoongClaw users, especially first-time installers
**Frequency:** Every first run
**Consequence if missing:** Poor first impression, confusion about how to get started, potential abandonment

### Submission checks
- [x] I searched existing issues and did not find the same request already tracked.
- [x] I used project-scoped wording and removed secrets or private data from examples.

### PR DESCRIPTION
**Pull request by gh-xj**
## Summary

- Problem:
  Bare `loongclaw` defaulted to `demo`, which exposed internal-looking bootstrap output on first run instead of guiding new users into onboarding.
- Why it matters:
  Issue `#400` is specifically about the first-run experience. The no-args entrypoint is a high-frequency path, and the old behavior created a poor first impression for new installs.
- What changed:
  Bare `loongclaw` now routes through a first-run resolver: it starts interactive onboarding when no config file exists, and shows a versioned quick-command welcome banner when a config file is already present. The follow-up review pass also tightened the explicit `welcome` path so it fails closed when the resolved config path is missing or is a directory, aligned the `welcome` help text with the actual `--config <path>` commands, and strengthened temp-dir uniqueness in the new tests.
- What did not change (scope boundary):
  `loongclaw demo` remains available as an explicit command, the onboarding flow itself was not redesigned, and this PR does not implement the browser-first / launcher direction tracked separately from `#400`.

## Linked Issues

- Closes #400
- Related #399

## Change Type

- [ ] Bug fix
- [x] Feature
- [ ] Refactor
- [ ] Documentation
- [x] Security hardening
- [ ] CI / workflow / release

## Touched Areas

- [ ] Kernel / policy / approvals
- [ ] Contracts / protocol / spec
- [x] Daemon / CLI / install
- [ ] Providers / routing
- [ ] Tools
- [ ] Browser automation
- [ ] Channels / integrations
- [ ] ACP / conversation / session runtime
- [ ] Memory / context assembly
- [x] Config / migration / onboarding
- [ ] Docs / contributor workflow
- [ ] CI / release / workflows

## Risk Track

- [x] Track A (routine / low-risk)
- [ ] Track B (higher-risk / policy-impacting)

## Validation

- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [x] `cargo test --workspace --locked`
- [x] `cargo test --workspace --all-features --locked`
- [x] Relevant architecture / dep-graph / docs checks for touched areas
- [x] Additional scenario, benchmark, or manual checks when behavior changed
- [x] If this changes config/env fallback, limits, or defaults: include before/after behavior and regression coverage for explicit path, fallback path, and boundary values
- [x] If tests mutate process-global env: document how state is restored or serialized

Commands and evidence:

```text
cargo test -p loongclaw-daemon first_run_entry -- --nocapture
- passed; covers config-missing, config-present, LOONGCLAW_CONFIG_PATH override, directory-path boundary, and explicit welcome failure cases.

cargo test -p loongclaw-daemon welcome_subcommand_help_advertises_first_run_shortcuts -- --nocapture
- passed; covers help-text alignment with --config <path> and LOONGCLAW_CONFIG_PATH guidance.

task verify
- passed; includes fmt, clippy, workspace tests, all-features tests, architecture checks, docs checks, and cargo deny.

Test env handling:
- the new daemon tests use `mvp::test_support::ScopedEnv` to restore process-global env state after each case.
- temp roots now include PID + AtomicU64 suffixes to avoid cross-test collisions.
```

Before / after behavior:

```text
Before:
- bare `loongclaw` defaulted to `demo`
- explicit `loongclaw welcome` printed a "configured and ready" banner even if the config path was missing or a directory

After:
- bare `loongclaw` starts onboarding when no config file exists
- bare `loongclaw` shows the welcome banner when a config file exists
- explicit `loongclaw welcome` errors with onboarding guidance if the resolved config path is not a file
- directory-valued config paths no longer route to welcome
```

## User-visible / Operator-visible Changes

- New users who run bare `loongclaw` are now sent directly into onboarding instead of the demo path.
- Existing users who run bare `loongclaw` now see a welcome banner with version info and quick commands.
- Operators using `loongclaw welcome` with a bad config path now get a clear error instead of a misleading success banner.

## Failure Recovery

- Fast rollback or disable path:
  Revert this PR to restore the old `demo` default.
- Observable failure symptoms reviewers should watch for:
  bare `loongclaw` incorrectly entering `Welcome` when the config path is missing or a directory, or `welcome` help drifting away from the actual banner commands.

## Reviewer Focus

- `crates/daemon/src/lib.rs`: default-entry routing, explicit `welcome` failure behavior, and env/path boundary handling.
- `crates/daemon/tests/integration/cli_tests.rs`: help-text contract for the new `welcome` command.

### CODE REVIEW

File: crates/daemon/src/lib.rs [hunk 1]
```diff
@@ -165,6 +165,11 @@ pub struct Cli {
 
 #[derive(Subcommand, Debug)]
 pub enum Commands {
+    #[command(
+        long_about = "Show the configured welcome banner and quick commands.\n\nquick commands:\n- loongclaw ask --message \"...\"\n- loongclaw chat\n- loongclaw doctor\n- loongclaw --help"
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply crates/daemon/src/lib.rs]: Fixed in `a2cbbe1`. `welcome` help now advertises `--config <path>` and explains the `LOONGCLAW_CONFIG_PATH` shortcut so the static help matches the rendered banner.

File: crates/daemon/src/lib.rs [hunk 2]
```diff
@@ -711,6 +716,65 @@ pub enum ValidateConfigOutput {
...
+        doctor_command,
+        CLI_COMMAND_NAME,
+    )
+}
+
+pub fn run_welcome_cli() -> CliResult<()> {
+    println!(
+        "{}",
+        render_welcome_banner(resolved_default_entry_config_path().as_path())
+    );
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply crates/daemon/src/lib.rs]: Fixed in `a2cbbe1`. `run_welcome_cli()` now fails closed when the resolved config path is not a file and points users back to `loongclaw onboard`.

File: tasks/todo.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- [x] Add tests for bare `loongclaw` no-args routing when config is missing vs present.
+- [x] Add tests for the welcome banner content and command hints.
+- [x] Implement a default entry resolver and a new `welcome` command path.
+- [x] Switch `main` to use the new default entry behavior while keeping `demo` explicit.
+- [x] Run targeted daemon tests during red/green cycles.
+- [x] Run `task verify`.
+
+## Progress
+
+- 2026-03-20: Created isolated product worktree `/Users/xj/github/loongclaw/loongclaw/.worktrees/issue-400-default-welcome` from `upstream/dev` for issue #400.
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply tasks/todo.md]: Fixed in `a2cbbe1`. The machine-specific path in `tasks/todo.md` is now redacted to `.worktrees/issue-400-default-welcome`, and a repo scan no longer finds any `/Users/xj` paths in this branch.

File: crates/daemon/src/lib.rs [hunk 4]
```diff
@@ -711,6 +716,65 @@ pub enum ValidateConfigOutput {
...
+        skip_model_probe: false,
+    }
+}
+
+pub fn resolve_default_entry_command() -> Commands {
+    if resolved_default_entry_config_path().exists() {
+        Commands::Welcome
+    } else {
+        default_onboard_command()
+    }
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply crates/daemon/src/lib.rs:747]: Fixed in `a2cbbe1`. Default-entry routing now uses `is_file()`, and the daemon tests cover the directory-path case so this does not regress.

File: crates/daemon/src/lib.rs [hunk 5]
```diff
@@ -753,6 +817,107 @@ pub async fn run_demo() -> CliResult<()> {
...
+        path::{Path, PathBuf},
+        time::{SystemTime, UNIX_EPOCH},
+    };
+
+    fn unique_temp_dir(prefix: &str) -> PathBuf {
+        let nanos = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("clock should be after epoch")
+            .as_nanos();
+        std::env::temp_dir().join(format!("{prefix}-{nanos}"))
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply crates/daemon/src/lib.rs]: Fixed in `a2cbbe1`. The test helper now includes the process id plus an `AtomicU64` counter, matching the stronger temp-path pattern already used in other daemon tests.

File: tasks/todo.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply tasks/todo.md]: Fixed in `a2cbbe1`. The worktree entry in `tasks/todo.md` is now redacted to `.worktrees/issue-400-default-welcome`.
